b2share/factory.py

```diff
@@ -1,192 +1,178 @@
-        rank=same;
-        fontsize = 20;
-        label = "B2SHARE";
-        style = "solid";
-
-        dispatcher [label="DispatcherMiddleware", shape="parallelogram"]
-        app [label="Top Flask\\napplication", shape="Mcircle"];
-        rest_app [label="REST API\\nApplication", shape="Mcircle"]
-        ui_files [label="UI Files", shape="folder"]
-        communities_views [label="b2share.modules.communities.views", shape="rectangle"]
-        schemas_views [label="b2share.modules.schemas.views", shape="rectangle"]
-        other_views [label="...", shape="rectangle"]
-    }
-    dispatcher -> app [label="endpoint != '/api/*'"];
-    app -> ui_files [label="serves"];
-    dispatcher -> rest_app [label="endpoint == '/api/*'"];
-    rest_app -> communities_views [label="endpoint in\\n['/api/communities/*', ...]"]
-    rest_app -> schemas_views [label="endpoint in\\n['/api/communities/<ID>/schemas/*', ...]"]
-    rest_app -> other_views [label="endpoint == '...'"]
-    }
-
-See Invenio and invenio_base module for more information regarding how Invenio
-applications are created. The *"UI Application"* in our case is a custom
-one, it does not match *"Invenio UI application"* which serves default
-Invenio UI.
-
-The ``*\*.views*`` are the ``views.py`` modules included in modules which
-contain the REST API definition. The requests are dispatched to the right
-view class using the Flask endpoints matching rules.
-
-**TODO**: Note that Invenio has evolved since :py:func:`~.create_app` was
-created. It will be necessary at some point to refactor it.
-"""
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Flask application factories for Invenio flavours."""
+
+from __future__ import absolute_import, print_function
 
 import os
 import sys
 
-from flask import Flask
+import pkg_resources
+
+from flask import current_app
+
 from invenio_base.app import create_app_factory
-from invenio_config import create_conf_loader
-from werkzeug.wsgi import DispatcherMiddleware
-from werkzeug.contrib.fixers import ProxyFix
+from invenio_base.wsgi import create_wsgi_factory, wsgi_proxyfix
+from invenio_base.signals import app_created, app_loaded
+from invenio_cache import BytecodeCache
+from invenio_config import create_config_loader
+
+from jinja2 import ChoiceLoader, FileSystemLoader
 
 from . import config
 
 env_prefix = 'B2SHARE'
 
-config_loader = create_conf_loader(config=config, env_prefix=env_prefix)
+invenio_config_loader = create_config_loader(
+    config=config, env_prefix=env_prefix
+)
+
+@app_created.connect
+def receiver_app_created(sender, app=None, **kwargs):
+    app.logger.debug("Application created")
+
+@app_loaded.connect
+def receiver_app_loaded(sender, app=None, **kwargs):
+    app.logger.debug("Application Loaded")
+    app.logger.debug("Instance Path: {}".format(app.instance_path))
+    check_configuration(app.config, app.logger)
```

Comment:
Mostly logging but NOTE check_configuration is handled with signals now and not syncronously as with V2: https://github.com/EUDAT-B2SHARE/b2share/blob/d846068c71853221ac53239c9efc56e3e795e6c3/b2share/factory.py#L150